Ea/aa ra (#1591)
hashicorp/terraform-website: content/source/docs/enterprise/before-installing/reference-architecture/azure.md

### Monitoring

There is not currently a full monitoring guide for Terraform Enterprise. The following pages include information relevant to monitoring:

- [Logging](../../admin/logging.html)
- [Diagnostics](../../support/index.html)
- [Reliability and Availability](../../system-overview/reliability-availability.html)

### Upgrades

See [the Upgrades section](../../admin/upgrades.html) of the documentation.

## High Availability - Failure Scenarios

Azure provides availability and reliability recommendations on [Azure reliability](https://azure.microsoft.com/en-us/features/reliability/). Working in accordance with those recommendations, the Terraform Enterprise Reference Architecture is designed to handle different failure
scenarios that have different probabilities. As the architecture evolves it will provide a
higher level of service continuity.

### Terraform Enterprise Server

> By utilizing an VM Scale Set, a Terraform Enterprise instance automatically recovers

Reviewer: ```suggestion
By utilizing an VM Scale Set, a Terraform Enterprise instance can automatically recover
```

    point for the infrastructure deployed in the secondary Azure
    Region.

## Active/Active Implementation Mode

### Overview

As stated previously, the _Active/Active_ implementation mode is an extension of the _Standalone_ implementation mode that increases the scalability and load capacity of the Terraform Enterprise platform. The same application runs on multiple Terraform Enterprise instances utilizing the same external services in a shared model. The primary architectural and implementation differences for _Active/Active_ are:

- It can only be run in the _External Services_ mode.

> - The second/alternate and additional nodes are active and processing work at all times.

Reviewer: ```suggestion
- The additional nodes are active and processing work at all times.
```

> - There is an addition to the existing external services which is a memory cache which is currently implemented with cloud native implementations of Redis. This is used for the processing queue for the application and has been moved from the individual instance to be a shared resource that manages distribution of work.

Reviewer: ```suggestion
- In an addition to the existing external services, there is a memory cache which is currently implemented with cloud native implementations of Redis. This is used for the processing queue for the application and has been moved from the individual instance to be a shared resource that manages distribution of work.
```

If you are considering a migration from a _Standalone_ implementation to _Active/Active_, it is straightforward and there is guidance available to assist with that effort. However, you should first make a determination if the move is necessary. The _Standalone_ mode is capable of handling significant load and the first paths to supporting higher load can be simply increasing the compute power in the existing implementation.  A discussion with your HashiCorp representatives may be warranted.

Also note that if your existing architecture does not already depict what is shown and discussed above, you will likely need to make adjustments to bring it into alignment. This could be either before or during the migration. Certain tenets of the reference architecture described here are highly recommended and potentially necessary to support _Active/Active_ mode such as load balancers and scaling groups.

### Infrastructure Diagram - Active/Active

![aws-aa-infrastructure-diagram](./assets/RA-TFE-AA-Azure-SingleRegion.png)

The above diagram shows the infrastructure components of an _Active/Active_ implementation at a high-level.

### Infrastructure Requirements

#### Active Nodes

> The diagram depicts two active nodes to be concise. Additional nodes can be added by simply launching another instance with the identical configuration that points to the same shared external services. The number and sizing of nodes should be based on load requirements and redundancy needs. Nodes should be deployed in alternate zones to accommodate zone failure.

Reviewer: Not always! Customers using node IP addresses in their configuration will need to make adjustments.

Author: altered language but not in detail
All changes reflected in GCP also. Committed under new branch and new PR
